metadata/src/main/java/org/apache/kafka/controller/SnapshotGenerator.java

```diff
@@ -74,56 +74,49 @@ String name() {
         this.batch = null;
         this.section = null;
         this.numRecords = 0;
-        this.numWriteTries = 0;
     }
 
     /**
      * Returns the epoch of the snapshot that we are generating.
      */
     long epoch() {
-        return writer.epoch();
+        return writer.lastOffset();
```

Comment:
Is this correct? Seems likely to cause confusion if it is.

Reply:
Yes but the names are not great. Updated the names of `SnapshotGenerator.epoch` and `SnapshotWriter.lastOffset` to `lastOffsetFromLog`. This should make it clear that the offset of the batches in the snapshots are independent of the last offset from the log that is included in the snapshot.